Realm/Realm/DatabaseTypes/RealmCollectionBase.cs

```diff
@@ -164,6 +164,12 @@ internal RealmResults<T> GetFilteredResults(string query, RealmValue[] arguments
             return new RealmResults<T>(Realm, resultsHandle, Metadata);
         }
 
+        internal RealmResults<TValue> GetFilteredResults<TValue>(string query, RealmValue[] arguments)
```

Comment:
I don't think it's a good idea to have this here in the base class, because it's not applicable to the other collections (list, sets)